pkg/collector/container.go

```diff
@@ -223,6 +235,36 @@ func getMetricsPort(c map[interface{}]interface{}) (int32, error) {
 	return int32(i64), nil
 }
 
+func getPrometheusExporterPort(c map[interface{}]interface{}) (int32, error) {
+	// we don't need to unmarshal the whole config, just follow the keys down to
+	// the prometheus endpoint.
+	type prometheusCfg struct {
+		Endpoint string
+	}
+	type exportersCfg struct {
+		Prometheus prometheusCfg
+	}
+	type cfg struct {
+		Exporters exportersCfg
+	}
+	var cOut cfg
+	err := mapstructure.Decode(c, &cOut)
+	if err != nil {
+		return 0, nil
+	}
+
+	_, port, err := net.SplitHostPort(cOut.Exporters.Prometheus.Endpoint)
```

Comment:
Does this also work when there are two prometheus exporters configured?

Reply:
Thanks for the feedback!